hack4impact-utk/compassion-ministries
# 185 create post eventeventidemail api server function

# Description
Defined `CreateEmailRequest` type.
Created `sendEventEmail()` server func to send an email to the email address of the volunteer associated with each eventVolunteer checked-in for an event.
Created POST API func at `src/app/api/events/[eventId]/email/route.ts` to call the `sendEventEmail()` function.
Set up new Google account "compassion.ministries.events@gmail.com" (credentials in VaultWarden).
Added OAuth2 credentials to `.env.local` (update your file from VaultWarden)

## Relevant issue(s)

#185 

## Questions

## Type of change


# Checklist:


# To Test:

Replace your `.env.local` with the updated version from the VaultWarden.
Create a new event by hitting the POST API endpoint at `/api/events/`:
```
{
    "name":"Email Test Event",
    "isRecurring":false,
    "startAt":"2024-01-01T15:00:00.000+00:00",
    "endAt":"2024-01-01T17:00:00.000+00:00",
    "eventRoles":["Food"],
}
```
Record the event ID that is returned.
Create a new volunteer associated with your email by hitting the POST API endpoint at `/api/volunteers/`:
```
{
    "firstName":"Email",
    "lastName":"Dummy",
    "email":"<your email address>", // replace with the email address you want to send the test email to
    "phoneNumber":"8675309",
    "address":"69420 Epic Street"
}
```
Record the volunteer ID that is returned.
Check your new volunteer in at your new event by hitting the POST API endpoint at `/api/events/[eventId]/check-in/`, replacing `[eventId]` with the ID string of the new event you created:
```
{
	"volunteer":"<your dummy volunteer id>", // replace with the volunteer ID of your new volunteer
	"role":"Food"
}
```
Send an email to the email address of the volunteer associated with each eventVolunteer for your new event by hitting the POST API endpoint at `/api/events/[eventId]/email/`, replacing `[eventId]` with the ID string of the new event you created:
```
{
    "subject": "test subject",
    "emailbody": "test email body"
}
```
Check your inbox and see if you got an email from compassion.ministries.events@gmail.com
